ygot/render_test.go

```diff
@@ -1660,33 +1660,65 @@ type invalidGoStructMap struct {
 func (*invalidGoStructMap) IsYANGGoStruct() {}
 
 type structWithMultiKey struct {
-	Map map[mapKey]*structMultiKeyChild `path:"foo"`
+	Map map[mapKey]*structMultiKeyChild `path:"foo" module:"rootmod"`
 }
 
 func (*structWithMultiKey) IsYANGGoStruct() {}
 
+type structWithMultiKeyInvalidModuleTag struct {
+	Map map[mapKey]*structMultiKeyChild `path:"foo/bar" module:"rootmod"`
+}
+
+func (*structWithMultiKeyInvalidModuleTag) IsYANGGoStruct() {}
+
+type structWithMultiKeyInvalidModuleTag2 struct {
+	Map map[mapKey]*structMultiKeyChild `path:"foo" module:"rootmod/rootmod"`
+}
+
+func (*structWithMultiKeyInvalidModuleTag2) IsYANGGoStruct() {}
+
+type structWithMultiKeyInvalidModuleTag3 struct {
+	Map map[mapKey]*structMultiKeyChild `path:"foo/bar" module:"rootmod/rootmod|rootmod"`
+}
+
+func (*structWithMultiKeyInvalidModuleTag3) IsYANGGoStruct() {}
+
+type structWithMultiKeyInvalidModuleTag4 struct {
+	Map map[mapKey]*structMultiKeyChild `path:"foo/bar" module:""`
+}
+
+func (*structWithMultiKeyInvalidModuleTag4) IsYANGGoStruct() {}
+
+type structWithMultiKeyInvalidModuleTag5 struct {
+	Map map[mapKey]*structMultiKeyChild `path:"foo/bar" module:"rootmod/rootmod2|rootmod"`
+}
+
+func (*structWithMultiKeyInvalidModuleTag5) IsYANGGoStruct() {}
+
 type mapKey struct {
 	F1 string `path:"fOne"`
 	F2 string `path:"fTwo"`
 }
 
 type structMultiKeyChild struct {
-	F1 *string `path:"config/fOne|fOne" shadow-path:"state/fOne|fOne"`
-	F2 *string `path:"config/fTwo|fTwo" shadow-path:"state/fTwo|fTwo"`
+	F1 *string `path:"config/fOne|fOne" module:"fmod/f1mod|f1mod" shadow-path:"state/fOne|fOne" shadow-module:"fmod/f1mod|f1mod"`
+	F2 *string `path:"config/fTwo|fTwo" module:"fmod/f2mod|f2mod" shadow-path:"state/fTwo|fTwo" shadow-module:"fmod/f2mod-shadow|f2mod-shadow"`
```

Comment:
Is this something that can actually happen? i.e., how does a compressed path end up being in two modules? Is this just to check that the `shadow-module` parsing works correctly?

Reply:
Interesting -- I hadn't thought about this case, it'd be weird, but could indeed happen.